include/nntile/base_types.hh

```diff
@@ -15,49 +15,376 @@
+//    INT32,
+//    INT16,
+//    INT8,
+//    BOOL,
+//    FP64,
+//    FP32,
+//    FP32_FAST_TF32,
+//    FP32_FAST_FP16,
+//    FP32_FAST_BF16,
+//    TF32,
+//    FP16,
+//    BF16,
+//    FP8_E4M3,
+//    FP8_E5M2
+//};
+//    constexpr char *name()
+//    {
+//        switch(value)
+//        {
+//            case INT64:
+//                return "int64";
+//                break;
+//            case INT32:
+//                return "int32";
+//                break;
+//            case INT16:
+//                return "int16";
+//                break;
+//            case INT8:
+//                return "int8";
+//                break;
+//            case BOOL:
+//                return "bool";
+//                break;
+//            case FP64:
+//                return "fp64";
+//                break;
+//            case FP32:
+//                return "fp32";
+//                break;
+//            case FP32_FAST_TF32:
+//                return "fp32_fast_tf32";
+//                break;
+//            case FP32_FAST_FP16:
+//                return "fp32_fast_fp16";
+//                break;
+//            case FP32_FAST_BF16:
+//                return "fp32_fast_bf16";
+//                break;
+//            case TF32:
+//                return "tf32";
+//                break;
+//            case FP16:
+//                return "fp16";
+//                break;
+//            case BF16:
+//                return "bf16";
+//                break;
+//            case FP8_E4M3:
+//                return "fp8_e4m3";
+//                break;
+//            case FP8_E5M2:
+//                return "fp8_e5m2";
+//                break;
+//            default:
+//                return "error";
+//        }
+//    }
+//};
 
-// Supported computational types
-//! Double precision alias
-using fp64_t = double;
-// class fp64_t {
-//     public:
-//         double _;
-//         double numpy_dtype;
-// };
-//! Single precision alias
-using fp32_t = float;
-// class fp32_t {
-//     public:
-//         float _;
-//         float numpy_dtype;
-// };
+//! Integer type for scalar values and indices outside StarPU buffers
+using Index = std::int64_t;
 
-//! Half precision FP16 mock type
-class fp16_t
+//! This type is meant for scalar values outside StarPU buffers
+using scal_t = float;
+
+//! NNTile type for std::int64_t inside StarPU buffers, Tile and Tensor
+class int64_t
 {
-private:
-    int16_t _;
+public:
+    using internal_t = std::int64_t;
+    using compat_t = std::int64_t;
+    static constexpr bool trivial_copy_from_compat = true;
+    internal_t value;
+    explicit constexpr int64_t(const compat_t &other):
+        value(other)
+    {
+    }
+    constexpr int64_t &operator=(const compat_t &other)
+    {
+        value = other;
+        return *this;
+    }
+    constexpr int64_t &operator=(const int64_t &other)
+    {
+        value = other.value;
+        return *this;
+    }
+    explicit constexpr operator compat_t() const
```

Comment:
why do we need this function? and what are the use cases?

Reply:
This is just a conversion function, that converts our wrapped type (e.g, nntile::fp16) into a compatible type basic type (float).